src/ctia/stores/es/crud.clj

```diff
@@ -117,32 +112,48 @@ It returns the documents with full hits meta data including the real index in wh
    es-params]
   (first (get-docs-with-indices conn-state [_id] es-params)))
 
-
 (defn ^:private prepare-opts
   [{:keys [props]}
    {:keys [refresh]}]
   {:refresh (or refresh
                 (:refresh props)
                 "false")})
 
+(s/defn bulk-schema
+  [Model :- (s/pred map?)]
+  (st/optional-keys
+   {:create [Model]
+    :index [Model]
+    :update [(st/optional-keys Model)]
+    :delete [s/Str]}))
+
+(s/defn ^:private prepare-bulk-doc
+  [{:keys [props]} :- ESConnState
+   mapping :- s/Keyword
+   doc :- (s/pred map?)]
+  (assoc doc
+         :_id (:id doc)
+         :_index (:write-index props)
+         :_type (name mapping)))
+
 (defn handle-create
   "Generate an ES create handler using some mapping and schema"
   [mapping Model]
   (let [coerce! (coerce-to-fn (s/maybe Model))]
     (s/fn :- (s/maybe [Model])
-      [{:keys [conn props] :as conn-state} :- ESConnState
+      [{:keys [conn] :as conn-state} :- ESConnState
```

Comment:
🔴 

How can you use `:-` in a non `s/defn` declaration?

Reply:
This is `s/fn` declaration :-)
http://plumatic.github.io/schema/schema.core.html#var-fn